core/src/main/java/org/apache/accumulo/core/metadata/schema/RootGcCandidatesJson.java

```diff
@@ -96,20 +98,6 @@ public Stream<String> stream() {
   }
 
   public String toJson() {
-    GSonData gd = new GSonData();
-    gd.candidates = candidates;
-    return GSON.toJson(gd);
-  }
-
-  public static RootGcCandidates fromJson(String json) {
-    GSonData gd = GSON.fromJson(json, GSonData.class);
-
-    Preconditions.checkArgument(gd.version == 1);
-
-    return new RootGcCandidates(gd.candidates);
-  }
-
-  public static RootGcCandidates fromJson(byte[] json) {
-    return fromJson(new String(json, UTF_8));
+    return GSON.toJson(this);
```

Comment:
I'm not so sure about this. The old code was robust against changes to this class. You added a warning above... but that's not as safe as enforcing in code the independence of the JSON serialization from changes to this class.